manual/src/cmds/native-debugger.etex

```diff
@@ -0,0 +1,289 @@
+
+OCaml defines its own calling convention that details how arguments are passed to functions, how values are returned and how registers are used. The calling convention is architecture specific and differs from the C calling convention. See [proc.ml] for specific details for your chosen architecture.
+
+The OCaml compiler generates line information that maps instructions back to their source program location (e.g., the instruction at address \texttt{0x00} originated from \texttt{myprogram.ml} line 42). This allows native debuggers to display the OCaml source code for the program being debugged and enables stepping through OCaml source code.
+
+Debuggers often need to view or modify the state of any function on the call stack, CFI allows the native debugger to find and inspect this state. CFI information is preserved across the OCaml / C boundary, so the OCaml runtime can be debugged.
+
+\subsection{ss:native-debugger-name-mangling}{Name Mangling}
+
+Name mangling describes how the OCaml compiler generates symbol names for OCaml language constructs. The format of these symbols is important for debuggers, performance, and observability tools to uniquely identify the source function for a symbol without referencing the original source code. Without source mappings, you often need to use mangled names to set breakpoints; otherwise they will appear in the information the native debugger displays. Understanding OCaml's name mangling is therefore essential when debugging OCaml programs. OCaml 5.1.1 uses the a name mangling scheme of \texttt{caml<MODULE_NAME>.<FUNCTION_NAME>_<NNN>}, where \texttt{NNN} is a randomly generated number. Before 5.1.1 the scheme is uses two underscores as the separator, e.g., \texttt{caml<MODULE_NAME>__<FUNCTION_NAME>_<NNN>}.
+
+\paragraph{Note:} For the Windows MSVC port (restored in OCaml 5.3), the scheme uses \texttt{\$} as the separator, e.g., \texttt{caml<MODULE_NAME>\$<FUNCTION_NAME>_<NNN>}.
+
+\subsection{ss:native-debugger-frame-pointers}{Frame Pointers}
+
+The OCaml native compiler supports generating frame pointers, which native debugger can use to walk the stack of function calls in a program. The frame pointer (also known as the base pointer) is a register (e.g., \texttt{\%rbp} on x86_64 or \texttt{x29} on ARM64) that points to the base of the current stack frame. The stack frame (also known as the activation frame or the activation record) refers to the portion of the stack allocated to a single function call. By saving the frame pointer along with the return address, the call stack for OCaml can be maintained. Using frame pointers only, without CFI enabled, it is possible to debug OCaml programs, however the experience is closer to debugging assembly and using DWARF with CFI is recommended. 
```

Comment:
```suggestion
The OCaml native compiler supports generating frame pointers, which native debugger can use to walk the stack of a program's function calls. The frame pointer (also known as the base pointer) is a register (e.g., \texttt{\%rbp} on x86_64 or \texttt{x29} on ARM64) that points to the base of the current stack frame. The stack frame (also known as the activation frame or the activation record) refers to the portion of the stack allocated to a single function call. By saving the frame pointer along with the return address, the call stack for OCaml can be maintained. Using frame pointers only, without CFI enabled, it is possible to debug OCaml programs, however the experience is closer to debugging assembly and using DWARF with CFI is recommended. 
```
Does this still make sense / is accurate if we remove that little prepositional phrase?